From d1df01781a2ebc8d90cff4bc482ce8db52a86ee7 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Mon, 19 Aug 2019 14:16:45 +0530
Subject: [PATCH 13/22] WIP-UPSTREAM: glib: Specify noreturn prototype for MSVC
 too

Equivalent upstream MR: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1078

This needs to be added before the prototype, unlike G_GNUC_NORETURN
which is after the prototype, so we can't add it to that. At least we
can mark `g_abort()` and `g_assert()` to avoid the C4715 warning:

https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4715?view=vs-2019
---
 glib/gtestutils.h | 3 +++
 glib/gutils.h     | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/glib/gtestutils.h b/glib/gtestutils.h
index 6c475e5..5bbc916 100644
--- a/glib/gtestutils.h
+++ b/glib/gtestutils.h
@@ -448,6 +448,9 @@ void    g_assertion_message             (const char     *domain,
                                          const char     *func,
                                          const char     *message);
 GLIB_AVAILABLE_IN_ALL
+#ifdef _MSC_VER
+__declspec(noreturn)
+#endif
 void    g_assertion_message_expr        (const char     *domain,
                                          const char     *file,
                                          int             line,
diff --git a/glib/gutils.h b/glib/gutils.h
index 560a84e..f8c3829 100644
--- a/glib/gutils.h
+++ b/glib/gutils.h
@@ -311,6 +311,9 @@ g_bit_storage_impl (gulong number)
 #  define g_abort() abort ()
 #else
 GLIB_AVAILABLE_IN_2_50
+#ifdef _MSC_VER
+__declspec(noreturn)
+#endif
 void g_abort (void) G_GNUC_NORETURN G_ANALYZER_NORETURN;
 #endif
 #endif
-- 
2.27.0.windows.1

